lib/logger/lib/logger.ex

```diff
@@ -990,24 +994,25 @@ defmodule Logger do
   end
 
   defp macro_log(level, data, metadata, caller) do
-    caller =
-      compile_time_application_and_file(caller) ++
-        case caller do
-          %{module: module, function: {fun, arity}, line: line} ->
-            [mfa: {module, fun, arity}, line: line]
-
-          _ ->
-            []
-        end
+    [{:file, file} | maybe_application] = compile_time_application_and_file(caller)
+
+    location =
+      case caller do
+        %{module: module, function: {fun, arity}, line: line} ->
```

Comment:
I think the only information missing here that could make this pattern match to fail would be `function:` being `nil`. Maybe it would be better to always pass at least `file` and `line` in `location` even when `mfa` can't be constructed? Especially after `file` was removed from `quoted_metadata` in ecd2bcebc9b26459085574f7d0ce224fe36049c7